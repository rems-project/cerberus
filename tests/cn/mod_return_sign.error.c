/* Modulo can be done with Integers of different signs and sizes
   but think about the return type carefully. */

// fails because it should return unsigned int
int different_sign (int x, unsigned int y)
/*@ requires y != 0u32;
    ensures return == x % y; @*/
{
    return x % y;
}