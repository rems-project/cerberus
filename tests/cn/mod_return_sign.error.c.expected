return code: 1
tests/cn/mod_return_sign.error.c:7:27: error: Type error
    ensures return == x % y; @*/
                          ^
Expression 'y' has type 'u32'.
I expected it to have type 'i32' because of tests/cn/mod_return_sign.error.c:7:23:
    ensures return == x % y; @*/
                      ^
